### PR TITLE
Improved code quality

### PR DESCRIPTION
**Pull request by wizard-28**
This PR improves code quality

### CODE REVIEW

**Comment by wizard-28** [comment]
```c
    while((readed = read(fd, buffer->chars + buffer->length, buffer->allocated - buffer->length)) == (buffer->allocated - buffer->length))
```
> Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20). 

[L377](https://github.com/LinusDierheimer/fastfetch/blob/ab2cb9e0e40b38fe5ceaa9a9d3ee0e31d02b4f2b/src/common/io.c#L377) in [io.c](https://github.com/LinusDierheimer/fastfetch/blob/master/src/common/io.c) returns this [warning](https://app.codacy.com/gh/wizard-28/fastfetch/file/57450828446/issues/source?bid=24014449&fileBranchId=24014449#l377)

---

```c
    instance->config.titleLength = (uint32_t) strlen(instance->state.passwd->pw_name) + 1 + strlen(hostname);
```
> Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126). 

[L11](https://github.com/LinusDierheimer/fastfetch/blob/ab2cb9e0e40b38fe5ceaa9a9d3ee0e31d02b4f2b/src/modules/title.c#L11) in [tile.c](https://github.com/LinusDierheimer/fastfetch/blob/master/src/modules/title.c) returns this [warning](https://app.codacy.com/gh/wizard-28/fastfetch/file/57450828445/issues/source?bid=24014449&fileBranchId=24014449#l11)

---

```c
    uint32_t endLength = strlen(end);
```
> Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126).

[L549](https://github.com/LinusDierheimer/fastfetch/blob/ab2cb9e0e40b38fe5ceaa9a9d3ee0e31d02b4f2b/src/util/FFstrbuf.c#L549) in [FFstrbuf.c](https://github.com/LinusDierheimer/fastfetch/blob/ab2cb9e0e40b38fe5ceaa9a9d3ee0e31d02b4f2b/src/util/FFstrbuf.c#L549) returns this [warning](https://app.codacy.com/gh/wizard-28/fastfetch/file/57450828442/issues/source?bid=24014449&fileBranchId=24014449#l549)

---

You may want to look into these.

File: README.md [hunk 1]
```diff
@@ -18,45 +18,45 @@ There are some presets defined for fastfech in [`presets`](presets), you can can
...
 ```
 
-##### DE versions
+### DE versions
 ```
 KDE Plasma, Gnome, Cinnamon, Mate, XFCE4, LXQt
 ```
 
-##### Terminal fonts:
+### Terminal fonts
```
Comment (LinusDierheimer) [inline README.md:59]: Those are by purpose, as i don't want the headers to show up too big on github. According to the [markdown documentation](https://www.markdownguide.org/basic-syntax/) this is allowed.

File: run.sh [hunk 2]
```diff
@@ -1,3 +1,4 @@
+#!/bin/env bash
```
Comment (LinusDierheimer) [inline run.sh:1]: I think this should be `#!/bin/env sh` as i don't use any bashism and therefore all posix complicant shell can run this script

File: tests/performance.sh [hunk 3]
```diff
@@ -1,3 +1,4 @@
+#!/bin/env bash
```
Comment (LinusDierheimer) [inline tests/performance.sh:1]: I think this should be `#!/bin/env sh` as i don't use any bashism and therefore all posix complicant shell can run this script

File: README.md [hunk 4]
```diff
@@ -18,45 +18,45 @@ There are some presets defined for fastfech in [`presets`](presets), you can can
...
-* [`libXFConf`](https://gitlab.xfce.org/xfce/xfconf): Needed for XFWM theme and XFCE Terminal font.  
-* [`libSQLite3`](https://www.sqlite.org/index.html): Needed for rpm package count.  
+*  [`libpci`](https://github.com/pciutils/pciutils): Needed for GPU output. _Should_ be available on every linux system.
+*  [`libX11`](https://gitlab.freedesktop.org/xorg/lib/libx11): Needed for resolution output
+*  [`libXrandr`](https://gitlab.freedesktop.org/xorg/lib/libxrandr): Needed for appending refresh rate to resolution output.
+*  [`libwayland-client`](https://wayland.freedesktop.org/): Better resolution performance + support for monitors with different refresh rates in wayland sessions.  
+*  [`libGIO`](https://developer.gnome.org/gio/unstable/): Needed for values that are only stored GSettings.
+*  [`libDConf`](https://developer.gnome.org/dconf/unstable/): Needed for values that are only stored in DConf + Fallback for GSettings.
+*  [`libXFConf`](https://gitlab.xfce.org/xfce/xfconf): Needed for XFWM theme and XFCE Terminal font.  
+*  [`libSQLite3`](https://www.sqlite.org/index.html): Needed for rpm package count.
```
Comment (LinusDierheimer) [inline README.md:28]: The [markdown documentation about unordered lists](https://www.markdownguide.org/basic-syntax/#unordered-lists) has also only one whitespace. Why a second?

**Comment by LinusDierheimer** [comment]
> Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20).

I use dynamic buffers, so i update them whenever more space is needed. No possible buffer overflow here. See full code:
```c
ssize_t readed;
while((readed = read(fd, buffer->chars + buffer->length, buffer->allocated - buffer->length)) == (buffer->allocated - buffer->length))
{
    buffer->length += (uint32_t) readed;
    ffStrbufEnsureCapacity(buffer, buffer->allocated * 2);
}
```
---
> instance->config.titleLength = (uint32_t) strlen(instance->state.passwd->pw_name) + 1 + strlen(hostname);

`pw_name` and the result of `hostname` are both guaranteed to be null terminated
---
> uint32_t endLength = strlen(end);

This is in a method only intended to be used with null terminated strings. If i have a use case where this is not ensured, creating a `bool ffStrbufEndsWithNS(const FFstrbuf* strbuf, uint32_t endLength, const char* end)` would be the right way. Like i did for a lot of other Strbuf methods.

File: README.md (hunk 4 above)
Comment (wizard-28) [reply README.md:28]: It violates [remark-lint-list-item-indent](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-list-item-indent)

File: README.md (hunk 1 above)
Comment (wizard-28) [reply README.md:59]: Okay, but is removing the `:` ok?

File: run.sh (hunk 2 above)
Comment (wizard-28) [reply run.sh:1]: right

File: tests/performance.sh (hunk 3 above)
Comment (wizard-28) [reply tests/performance.sh:1]: right

File: README.md (hunk 1 above)
Comment (LinusDierheimer) [reply README.md:59]: TBH i don't care as long as it is consistent

Comment (wizard-28) [reply README.md:59]: Okay then

Comment (LinusDierheimer) [inline README.md:59]: Still waiting for the headers. removal of the colons is ok

Comment (wizard-28) [reply README.md:59]: Oh, okay!

Comment (wizard-28) [reply README.md]: done

**Review by LinusDierheimer — APPROVED** [review]
Seems fine, merging this.

**Pull request merged**